src/manager.py

```diff
@@ -0,0 +1,227 @@
+        last_time_event = last_time.replace('Z', '+00:00')
+        last_time_event = datetime.fromisoformat(last_time_event)
+        last_time_event = last_time_event + timedelta(seconds=1)
+        last_time_event = last_time_event.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        return last_time_event
+
+    def read_config(self):
+        """
+        Reads configuration information from a YAML file.
+        """
+        with open(self.config_file) as f:
+            data = yaml.safe_load(f)
+
+        if not self.check_keys(data):
+            return False
+
+        self.logzio_url = data['logzio']['url']
+        self.logzio_token = data['logzio']['token']
+        self.jumpcloud_url = "https://api.jumpcloud.com/insights/directory/v1/events"
+        self.last_time_event = data['jumpcloud_api']['start_date']
+        if self.last_time_event is None or 'start_date' in data['jumpcloud_api']:
+            self.last_time_event = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        elif self.is_valid_format(str(self.last_time_event)):
+            self.last_time_event = data['jumpcloud_api']['start_date']
+        self.jumpcloud_token = data['jumpcloud_api']['credentials']['token']
+        self.headers = {
+            "accept": "application/json",
+            "x-api-key": self.jumpcloud_token,
+            "content-type": "application/json"
+        }
+
+        self.jumpcloud_time_interval = data['jumpcloud_api']['settings']['time_interval']
+        if self.jumpcloud_time_interval is None or 'time_interval' not in data['jumpcloud_api']['settings']:
+            self.jumpcloud_time_interval = 5
+        return True
+
+
+    def write_last_time_to_file(self,last_time_event:str) -> None:
+        with open(self.last_start_dates_file, 'w') as f:
+            f.write(last_time_event)
+
+    class logzio_error(Exception):
+        pass
+    def send_events_to_logzio(self, events) -> None:
+        """
+        Sends a list of events to Logz.io.
+
+        :param events: The list of events to send to Logz.io.
+        """
+        self.logzio_shipper = LogzioShipper(self.logzio_url, self.logzio_token)
+        try:
+            if len(events) != 0:
+                logger.info("Events number: {}".format(len(events)))
+                for event in events:
+                    if "changes" in event:
+                        event["changes"] = self.changesValueToJSON(event["changes"])
+                        event_str = json.dumps(event)
+                    else:
+                        event_str = json.dumps(event)
+                    self.logzio_shipper.add_log_to_send(event_str)
+
+                self.logzio_shipper.send_to_logzio()
+                self.last_time_event = events[-1]['timestamp']
+            else:
+                logger.info("No events")
+            self.last_time_event = self.last_time_plus_sec(self.last_time_event)
+            self.write_last_time_to_file(self.last_time_event)
+        except Exception as e:
```

Comment:
you need to catch different exceptions from the logio_shipper. if you get 400, 401 or url error you want to exit the application. if you get 500 or connection error you want to try again in the next iteration.

Reply:
Can you give me an example I can do it?
For example, if I got error 500 from logz.io.